### DISCUSSION TITLE
simple daytime schedule

### DISCUSSION THREAD

**Discussion by gualtierogrey** [savonet/liquidsoap#3897] [User Support]
Hi, just a simple question. Trying to create a simple schedule for a different playlist at fixed time of the day, some day of the week. I found this reference but dated 2020 so not sure if this is still working as docs reference link is dead

https://github.com/savonet/liquidsoap/issues/1121

does the {1w and 1h-7h} still work?

here is part of my code

radio = switch([({ 6h-10h }, morning),
				({ 10h-17h }, afternoon),
                                ({ 3w and 17h-19h }, pl1),
                                ({ 4w and 17h-19h }, pl2),
                                ({ 5w and 17h-19h }, pl3),
                                ({ 6w and 17h-19h }, pl4),
				({ 19h-23h }, evening),
				({ 23h-3h }, night),
				({ 3h-6h }, dawn)])

**Comment by gAlleb** [comment]
Hi! Everything is intact. Please see page 100 https://www.liquidsoap.info/book/book.pdf

```
radio = switch([({ 6h-10h }, morning),
({ 10h-17h }, afternoon),
({ (3w) and 17h-19h }, pl1),
({ (4w) and 17h-19h }, pl2),
({ (5w) and 17h-19h }, pl3),
({ (6w) and 17h-19h }, pl4),
({ 19h-23h }, evening),
({ 23h-3h }, night),
({ 3h-6h }, dawn)])
```

This should work.

P.S. I doubt that this is an issue.

**Reply by vitoyucepi** [reply]
Hi all,
@gAlleb, you're absolutely right, everything should work.
I think `10h` and `3w` are some kind of syntactic sugar. The expression is evaluated to a boolean value. Try `print(3w)`.

Writing a custom function using `time.local` or `time.utc` is the way to go if you want to define more complex behavior.
```ruby
def a()
  let {min} = time.local()
  min < 5
end

switch([
  (a, sine()),
  ({true}, blank())
])
```

**Reply by gAlleb** [reply]
Hi, @vitoyucepi! Nice. Also some more complex behavior, say you want to set the date explicitly, could be achieved with functions like:

```ruby

def my_own_date_range() =
    # 2024-11-15 01:00:00
    range_start = 1731632400.
    # 2024-12-31 23:59:59
    range_end = 1735678799.
    current_time = time()
    result = (range_start <= current_time and current_time <= range_end)
    result
end

radio = switch(track_sensitive=true, [
({ my_own_date_range() }, pl1), ({true}, radio)
])

```

**Reply by vitoyucepi** [reply]
> `range_start = 1731632400.`

`time.make()`

**Reply by vitoyucepi** [reply]
> `({ my_own_date_range() }, pl1), ({true}, radio)`

`(my_own_date_range, pl1), ({true}, radio)`
`{}` is a function, that takes no arguments.

**Reply by gAlleb** [reply]
even better
Thank you
